### PR TITLE
Upload test results to test observability server

### LINKED ISSUE #432: Upload CI test results to test observability server
## Background

I’ve noticed that we seem to have occasional test failures when running our tests in CI. We should identify the frequently-failing tests so that we can understand and resolve their failures.

In other Ably projects, we've been using the [test observability server](https://test-observability.herokuapp.com/repos) to identify frequently-failing tests. This server is described further in https://github.com/ably/test-observability#readme.

## What do to

- Find out how to get our tests to emit JUnit format results
- Upload these results to the test observability server, as described [here](https://github.com/ably/test-observability#how-to-upload-results-to-it) (with preference for using https://github.com/ably/test-observability-action if possible).

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
Resolves #432.

Here's an example of a successful upload (with a deliberate test failure): https://test-observability.herokuapp.com/repos/ably/ably-asset-tracking-swift/uploads/719f4d4f-e5cd-4811-888e-8179f443873f

### CODE REVIEW

**Comment by lawrence-forooghian** [comment]
Here's an example of a successful upload (with a deliberate test failure): https://test-observability.herokuapp.com/repos/ably/ably-asset-tracking-swift/uploads/719f4d4f-e5cd-4811-888e-8179f443873f

File: Scripts/test.sh [hunk 1]
```diff
@@ -23,4 +23,16 @@ struct Secrets {
 }
 EOF
 
-set -o pipefail && xcodebuild test -scheme "ably-asset-tracking-swift-Package" -destination 'platform=iOS Simulator,name=iPhone 12' | xcpretty
+set +e
```
Comment (QuintinWillison) [inline Scripts/test.sh:26]: Perhaps some commentary above this line to explain why you need to do this. Perhaps also remove the blank line below this line, and also the blank line above the matching `set -e` lower down, in order to visually group them. 🤔

**Review by QuintinWillison — APPROVED** [review]
Looks good to me. Just one minor tweak that might enhance maintainability.

File: Scripts/test.sh (hunk 1 above)
Comment (QuintinWillison) [reply Scripts/test.sh:30]: Lovely. Thanks!

**Pull request merged**